### PR TITLE
Visualization of vibration and POGO effects for CSM virtual cockpit

### PR DESCRIPTION
**Pull request by ggalfi**
Improvement of the originally G-load driven shaking of the virtual cockpit. Now the visualized vibration is a superposition of three factors:
- G-load based offset of the viewpoint (practically head movement due to the G-load caused by thrust and aerodynamic forces)
- Noise-like viewpoint jostling of viewpoint depending on  altitude (reflected shockwaves); on dynamic pressure (aerodynamic vibration) and on thruster activity (engine vibration carried to the cockpit by the structure)
- longitudinal oscillation (POGO effect) depending on the phase of the launch.
The simulation of noise and longitudinal oscillation is based on Mission Reports, Launch Vehicle Flight Evaluation Reports and other sources (the most useful ones were Saturn V Launch Vehicle Flight Evaluation Report - AS-506, Apollo 11 Mission and the formal and informal crew reports in Apollo Flight Journal).

This model of vibration is visual only and has no effect on other parts of the simulation.

### CODE REVIEW

File: Orbitersdk/samples/ProjectApollo/src_csm/saturn.cpp [hunk 1]
```diff
@@ -2692,24 +2694,158 @@ void Saturn::GenericTimestep(double simt, double simdt, double mjd)
...
-	//
+	// Visualizing vibration
+	// This simulation of the vibration environment in the CM is mostly based on
+	// Saturn V Launch Vehicle Flight Evaluation Report - AS-506, Apollo 11 Mission
+	double dynpress = GetDynPressure();
+	VECTOR3 vAccel, vWeight, vAngAcc;
+	GetForceVector(vAccel);
+	GetWeightVector(vWeight);
+	
+	GetAngularAcc(vAngAcc);
```
Comment (indy91) [inline Orbitersdk/samples/ProjectApollo/src_csm/saturn.cpp:2705]: vAngAcc seems to not be used by anything.

**Comment by indy91** [comment]
Otherwise the pull request is very good. Maybe adds a bit too much code that runs on every timestep even if there is no maneuver or flight through the atmosphere going on, but it's not much worse than before. Looking forward to trying some launches and reentries with it.

File: Orbitersdk/samples/ProjectApollo/src_csm/saturn.cpp (hunk 1 above)
Comment (ggalfi) [reply Orbitersdk/samples/ProjectApollo/src_csm/saturn.cpp:2705]: Oops, I forgot to remove the remnants of some experimenting with the angular acceleration. My goal was to replicate the sideways zig-zagging of the rocket during the S-IC burn (see e.g. some comments of Bill Anders or Mike Collins on their Saturn V experience). But it turned  out that in the simulation the F1 gimbaling is way too smooth to cause any significant angular acceleration. Maybe the real-life gimbal was much coarser and more hectic or the more rearward CG caused larger swaying at the top of the rocket, I don't know. Anyway it didn't make any perceptible difference, so I decided to get rid of the code for this. I just simply overlooked the query of angular acceleration, but it is indeed some obsolete code. Can you ignore this line during merge or you want me to make a new commit and PR with vAngAcc stuff is removed?

Comment (indy91) [reply Orbitersdk/samples/ProjectApollo/src_csm/saturn.cpp:2705]: There are three lines where vAngAcc appears (Its declaration, GetAngularAcc and one commented) so it's probably best if you make a small commit to remove it. After that I see no obstacle anymore for getting this PR merged.

Comment (n7275) [reply Orbitersdk/samples/ProjectApollo/src_csm/saturn.cpp:2705]: @ggalfi, just so you know. If you make that change that @indy91 requested, and just commit and push in this vc branch, that's all you should need do. No need to make a new branch or pull request.

Comment (ggalfi) [reply Orbitersdk/samples/ProjectApollo/src_csm/saturn.cpp]: > No need to make a new branch or pull request.
Good to know, I almost removed this PR, when I've seen your comment.

Comment (ggalfi) [reply Orbitersdk/samples/ProjectApollo/src_csm/saturn.cpp]: I have done the code cleanup, and while I discovered another minor bug, I adjusted the logic  a little bit. I've tested the new code in the usual non-weightless (weightful ;) ) flight phases.

Comment (indy91) [reply Orbitersdk/samples/ProjectApollo/src_csm/saturn.cpp]: I'll go ahead and merge the PR then. Awesome work!

**Review by indy91 — APPROVED** [review]

**Pull request merged**